api/layer/object.go

```diff
@@ -268,7 +274,14 @@ func (n *layer) headLastVersionIfNotDeleted(ctx context.Context, bkt *data.Bucke
 		return nil, err
 	}
 	objInfo := objectInfoFromMeta(bkt, meta)
-	if err = n.objCache.PutObject(objInfo); err != nil {
+
+	extObjInfo := &data.ExtendedObjectInfo{
+		ObjectInfo:  objInfo,
+		NodeVersion: node,
+		IsLatest:    true,
```

Comment:
It seems this field is used only in listing operation so I suggest don't set it here.